go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -380,18 +400,10 @@ func (n *NodeController) startNamespaceWatch(wf *factory.WatchFactory) error {
 				if len(oldTunCookie) > 0 {
 					n.deleteFlowsByCookie(oldTunCookie)
 				}
-				pods, err := wf.GetPods(nsNew.Namespace)
-				if err != nil {
-					klog.Errorf("failed to get pods for NS update in hybrid overlay: %v", err)
-				}
-				for _, pod := range pods {
-					// we need to ignore learning cookie flow from table 20 during this pod update
-					// this is because the VTEP/GW may have changed, and now we need to get rid of the
-					// corresponding table 20 flow and not cache it
-					if err := n.addOrUpdatePod(pod, true); err != nil {
-						klog.Warningf("failed to handle pod %v update: %v", pod, err)
-					}
+				if err := n.updateNamespace(nsNew); err != nil {
```

Comment:
what about on namespace add, dont we also need to copy the annotations to pod?

Reply:
fixed